components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -435,6 +435,14 @@ void RewardsServiceImpl::CheckPreferences() {
     // utility process.
     StartEngineProcessIfNecessary();
   }
+
+  // If Rewards is enabled and the user has a linked account, then ensure that
+  // the "search ads enabled" pref has the appropriate default value.
+  if (prefs->GetBoolean(prefs::kEnabled) &&
+      !prefs->GetString(prefs::kExternalWalletType).empty() &&
+      !prefs->GetUserPrefValue(brave_ads::prefs::kOptedInToSearchResultAds)) {
+    prefs->SetBoolean(brave_ads::prefs::kOptedInToSearchResultAds, false);
+  }
```

Comment:
@aseren can you please check this logic, because we need to consider the business logic if a user has not joined Rewards and the ads library is always running. @zenparsing I will discuss with @aseren and we should catch up before merging. Thanks

Reply:
We also need to consider iOS changes.

Reply:
@zenparsing are we calling `!prefs->GetUserPrefValue(brave_ads::prefs::kOptedInToSearchResultAds)` to check if the pref exists, if so maybe we could use `HasPrefPath` which returns false if the pref has never been set.